### ISSUE TITLE
[Bug]for published checkpoint/config, vision transformer (224) inference, key missing,  even for cpu, different run will give different result

### ISSUE THREAD

**Issue by yige2018** [open-mmlab/mmclassification#820]
推荐使用英语模板 Bug report，以便你的问题帮助更多人。

### 描述 bug
简单地描述一下遇到了什么 bug
vision transformer vit-base-p16_3rdparty_pt-64xb64_in1k-224_20210928-02284250.pth + config vit-base-p16_pt-64xb64_in1k-224.py,
even for cpu, during inference, it reports key missing such as head.layer.head.weights. output result is different for each run. I tried to use test.py in the tools directory, it still have the same issue. Basically, for 224 input, all of them has issue, for 384 input, it works fine.
[填写这里]

### 复现流程
在命令行中执行的详细操作
 python tools/test.py --out vit1.pkl --device CPU configs/vision_transformer/vit-base-p16_pt-64xb64_in1k-224.py checkpoints/vit-base-p16_3rdparty_pt-64xb64_in1k-224_20210928-02284250.pth

after two run, I will diff the vit1.pkl and vit2.pkl, the result are different.
```shell
[填写这里]

The model and loaded state dict do not match exactly

missing keys in source state_dict: head.layers.pre_logits.weight, head.layers.pre_logits.bias, head.layers.head.weight, head.layers.head.bias
/nfs/yi/mmclassification/tools/test.py:161: UserWarning: Class names are not saved in the checkpoint's meta data, use imagenet by default.
  warnings.warn('Class names are not saved in the checkpoint\'s '
[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 5/5, 3.9 task/s, elapsed: 1s, ETA:     0s
dumping results to vit1.pkl
```

### 相关信息
1. `pip list | grep "mmcv\|mmcls\|^torch"` 命令的输出
[填写这里]

mmcls              0.22.0      /nfs/ssestorage01/sse.home.4/yge1/mmclassification
mmcv-full          1.4.8
mmdeploy           0.4.0       /nfs/ssestorage01/sse.home.4/yge1/mmclassification/mmdeploy

torch              1.11.0
2. 如果你修改了，或者使用了新的配置文件，请在这里写明
```python
[填写这里]
```
3. 如果你是在训练过程中遇到的问题，请填写完整的训练日志和报错信息
[填写这里]
4. 如果你对 `mmcls` 文件夹下的代码做了其他相关的修改，请在这里写明
[填写这里]

### 附加内容
任何其他有关该 bug 的信息、截图等

[填写这里]

**Comment by Ezra-Yu** [comment]
Do you mean this link? It is a pre-trained model in imagnet21k. the head of pre-train and fine-tune is different. So the head weights in your inferences are random.
Sorry that the checkpoint name makes you confused.
![image](https://user-images.githubusercontent.com/18586273/167234014-43cb8e97-6976-4eda-9130-d0b65c77315f.png)

**Comment by yige2018** [comment]
@Ezra-Yu Thanks for the reply. so, Could you suggest any existing fine-tuned model we can use for input 224?

**Comment by Ezra-Yu** [comment]
We don't have it yet, you need to convert one by yourself.

Why do you want such a pre-train model?

**Comment by yige2018** [comment]
in the hugging face, I saw checkpoint vit-base-patch16-224, is there any tool I can convert it to checkpoint which can use directly in the mmclassification? It gave me error if I directly used now.
@Ezra-Yu To answer your question, we are more focus on device level performance analysis, not really on the algorithm itself.
but if there is no other easy way, I will try to fine tune it.

**Comment by Ezra-Yu** [comment]
If you don't care about the accuracy, use the [publish_model.py](https://github.com/open-mmlab/mmclassification/blob/master/tools/convert_models/publish_model.py) tool to fix the checkpoint directly.

If there is a need for precision, you can refer to the [tool](https://github.com/open-mmlab/mmclassification/blob/master/tools/convert_models/twins2mmcls.py) to convert twins. Torch's checkpoint uses orderDict to save weight, just convert the key to the corresponding name.

**Comment by yige2018** [comment]
@Ezra-Yu Thanks, I just try publish_model.py, after converting, on CPU, I run twice, the result are still different. 
during running I saw "missing keys in source state_dict: head.layers.pre_logits.weight, head.layers.pre_logits.bias, head.layers.head.weight, head.layers.head.bias"
Here is the command for me to use to convert:

python publish_model.py ../../checkpoints/vit-base-p16_3rdparty_pt-64xb64_in1k-224_20210928-02284250.pth ../../checkpoints/vit-base-p16_3rdparty_pt-64xb64_in1k-224_20210928-02284250_final.pth

**Comment by Ezra-Yu** [comment]
Sorry, my bad, you can use below script( as $MMClassification/fix_vit.py).

```fix_vit.py
# Copyright (c) OpenMMLab. All rights reserved.
from argparse import ArgumentParser

from mmcls.apis import init_model
import torch


def main():
    parser = ArgumentParser()
    parser.add_argument('config', help='Config file')
    parser.add_argument('src', help='src Checkpoint file')
    parser.add_argument('dist', help='dist Checkpoint file')
    args = parser.parse_args()

    # build the model from a config file and a checkpoint file
    model = init_model(args.config, args.src, device="cpu")
    # test a single image
    torch.save(model.state_dict(), args.dist)


if __name__ == '__main__':
    main()
```


then run this command get convert weight:
```
python fix_vit.py ./configs/vision_transformer/vit-base-p16_pt-64xb64_in1k-224.py vit-base-p16_in21k-pre-3rdparty_ft-64xb64_in1k-384_20210928-98e8652b.pth convert_vit.pth
```

run command will get a demo result:
```
python demo/image_demo.py demo/demo.JPEG ./configs/vision_transformer/vit-base-p16_pt-64xb64_in1k-224.py convert_vit.pth --device cpu
```

**Comment by yige2018** [comment]
@Ezra-Yu Thanks! I saw consistent result.
